sql/src/test/java/org/apache/druid/sql/calcite/run/SqlResultsTest.java

```diff
@@ -102,6 +104,20 @@ public void testCoerceNestedArrays()
     assertCoerced(nestedList, nestedArray, true);
   }
 
+  @Test
+  public void testCoerceOfArrayOfPrimitives()
+  {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      SqlResults.coerce(mapper, new SqlResults.Context(null, false, false), new byte[0],
+                        SqlTypeName.BIGINT, "testFieldName"
+      );
+      Assert.fail("Should throw an exception");
+    }
+    catch (Exception e) {
+      Assert.assertEquals("Cannot coerce field [testFieldName] of value with class [[B] to BIGINT", e.getMessage());
```

Comment:
can we also more friendly type names such as `binary` instead of `[[B`?

Reply:
We could add the mapping of primitive array class names to actual names. Following list is standard to Strings for primitive java arrays classnames.
[Z = boolean
[B = byte
[S = short
[I = int
[J = long
[F = float
[D = double
[C = char
[L = any non-primitives(Object)